pkg/buildconfig/constants.go

```diff
@@ -0,0 +1,10 @@
+// Code generated by pkg/codegen/config/main.go. DO NOT EDIT.
+// Package buildconfig contains a set of exported constants that represent configuration variables of Rancher at build-time.
+package buildconfig
+
+const (
+	CspAdapterMinVersion = "103.0.0+up3.0.0"
+	DefaultShellVersion  = "rancher/shell:v0.1.22"
+	FleetVersion         = "103.1.0+up0.9.0-rc.5"
+	WebhookVersion       = "103.0.0+up0.4.0"
+)
```

Comment:
This doesn't have to be generated via a template. We should leave all of these constants as empty strings and have the value be passed in as a build arg: https://blog.alexellis.io/inject-build-time-vars-golang/

This is how version is injected, see https://github.com/rancher/rancher/blob/58bfcb19605d51bf24f0aa5bf1d60a68b1ae2a27/pkg/version/version.go#L12-L15 and https://github.com/rancher/rancher/blob/58bfcb19605d51bf24f0aa5bf1d60a68b1ae2a27/scripts/build-server#L29-L30

Reply:
If we take this approach, I believe we can entirely eliminate the `go generate` and `buildconfig` logic in favor of just a change to the build script.

Reply:
When we met within the team to discuss this design, we did have an idea of a build-time variable injection. But we decided against it. I don't recall all the reasons (almost 6 months have passed). Perhaps @MbolotSuse or @KevinJoiner could chime in here.

One issue was that devs may not necessarily be building Rancher using the build script, they just use `go build`. Without explicit injection, the variables will remain initialized with placeholder values, which may produce invalid behavior at runtime.

Reply:
FWIW, I still think build-time injection is a better approach here since go generating a constant file seems too heavyweight to me and it matches the pattern we use to build the docker image using build args.

Reply:
I don't think we should add build-time injection for required variables since it complicates the usage of a lot of native go commands such as `go run`, `go test`, `go install` and `go build`. build-time injection is primarily used for things like commit/version tags so you know when the build was created, but these values are required at runtime for rancher to work.

On the other hand, we already use `go generate` to create constant files so we are not breaking any patterns. 
I also would not consider `go generate` heavy. I think there may be bias towards Rancher's Controller generation which is heavy, but that is a Rancher problem not a `go generate` problem.